lib/rubygems/test_case.rb

```diff
@@ -1161,35 +1161,41 @@ def util_zip(data)
     Zlib::Deflate.deflate data
   end
 
-  def util_set_RUBY_VERSION(version, patchlevel = nil, revision = nil)
+  def util_set_RUBY_VERSION(version, patchlevel = nil, revision = nil, description = nil)
     if Gem.instance_variables.include? :@ruby_version or
        Gem.instance_variables.include? '@ruby_version' then
       Gem.send :remove_instance_variable, :@ruby_version
     end
 
-    @RUBY_VERSION    = RUBY_VERSION
-    @RUBY_PATCHLEVEL = RUBY_PATCHLEVEL if defined?(RUBY_PATCHLEVEL)
-    @RUBY_REVISION   = RUBY_REVISION   if defined?(RUBY_REVISION)
+    @RUBY_VERSION     = RUBY_VERSION
+    @RUBY_PATCHLEVEL  = RUBY_PATCHLEVEL  if defined?(RUBY_PATCHLEVEL)
+    @RUBY_REVISION    = RUBY_REVISION    if defined?(RUBY_REVISION)
+    @RUBY_DESCRIPTION = RUBY_DESCRIPTION if defined?(RUBY_DESCRIPTION)
 
     Object.send :remove_const, :RUBY_VERSION
-    Object.send :remove_const, :RUBY_PATCHLEVEL if defined?(RUBY_PATCHLEVEL)
-    Object.send :remove_const, :RUBY_REVISION   if defined?(RUBY_REVISION)
+    Object.send :remove_const, :RUBY_PATCHLEVEL  if defined?(RUBY_PATCHLEVEL)
+    Object.send :remove_const, :RUBY_REVISION    if defined?(RUBY_REVISION)
+    Object.send :remove_const, :RUBY_DESCRIPTION if defined?(RUBY_DESCRIPTION)
 
-    Object.const_set :RUBY_VERSION,    version
-    Object.const_set :RUBY_PATCHLEVEL, patchlevel if patchlevel
-    Object.const_set :RUBY_REVISION,   revision   if revision
+    Object.const_set :RUBY_VERSION,     version
+    Object.const_set :RUBY_PATCHLEVEL,  patchlevel  if patchlevel
+    Object.const_set :RUBY_REVISION,    revision    if revision
+    Object.const_set :RUBY_DESCRIPTION, description if description
   end
 
   def util_restore_RUBY_VERSION
     Object.send :remove_const, :RUBY_VERSION
-    Object.send :remove_const, :RUBY_PATCHLEVEL if defined?(RUBY_PATCHLEVEL)
-    Object.send :remove_const, :RUBY_REVISION   if defined?(RUBY_REVISION)
+    Object.send :remove_const, :RUBY_PATCHLEVEL  if defined?(RUBY_PATCHLEVEL)
+    Object.send :remove_const, :RUBY_REVISION    if defined?(RUBY_REVISION)
+    Object.send :remove_const, :RUBY_DESCRIPTION if defined?(RUBY_DESCRIPTION)
 
-    Object.const_set :RUBY_VERSION,    @RUBY_VERSION
-    Object.const_set :RUBY_PATCHLEVEL, @RUBY_PATCHLEVEL if
+    Object.const_set :RUBY_VERSION,     @RUBY_VERSION
+    Object.const_set :RUBY_PATCHLEVEL,  @RUBY_PATCHLEVEL  if
       defined?(@RUBY_PATCHLEVEL)
-    Object.const_set :RUBY_REVISION,   @RUBY_REVISION   if
+    Object.const_set :RUBY_REVISION,    @RUBY_REVISION    if
       defined?(@RUBY_REVISION)
+    Object.const_set :RUBY_DESCRIPTION, @RUBY_DESCRIPTION if
+      defined?(@RUBY_DESCRIPTION)
```

Comment:
Sorry for the big diff in this method, this is the gotcha of this aligment style: keeping it obscures the point of commits.